src/file.cr

```diff
@@ -795,6 +795,7 @@ class File < IO::FileDescriptor
   # See `self.new` for what *mode* can be.
   def self.write(filename : Path | String, content, perm = DEFAULT_CREATE_PERMISSIONS, encoding = nil, invalid = nil, mode = "w")
     open(filename, mode, perm, encoding: encoding, invalid: invalid) do |file|
+      file.sync = true
```

Comment:
ok, makes sense. was only thinking of the case when content was a string and no encoding was supplied. Disabled buffering only for Bytes and IO now.

Reply:
I'm not entirely convinced about `IO` because there's just no way to tell how that IO is gonna spit out data. Might be okay, maybe I'm just overly cautious. I suppose that would rather be an edge case that you could handle a different way. This method is intended for the most common use cases.

I suppose we could perhaps add a special case for `String`?
```cr
when String
  file.sync = true
  file.print(content)  
```